src/main/java/io/nats/client/JetStreamManagement.java

```diff
@@ -324,6 +325,48 @@ public interface JetStreamManagement {
      */
     MessageInfo getNextMessage(String streamName, long seq, String subject) throws IOException, JetStreamApiException;
 
+    /**
+     * Request a batch of messages using a {@link MessageBatchGetRequest}.
+     * <p>
+     * This API is currently EXPERIMENTAL and is subject to change.
+     *
+     * @param streamName the name of the stream
+     * @param messageBatchGetRequest the request details
+     * @return a list containing {@link MessageInfo}
+     * @throws IOException covers various communication issues with the NATS
+     *         server such as timeout or interruption
+     * @throws JetStreamApiException the request had an error related to the data
+     */
+    List<MessageInfo> fetchMessageBatch(String streamName, MessageBatchGetRequest messageBatchGetRequest) throws IOException, JetStreamApiException;
+
+    /**
+     * Request a batch of messages using a {@link MessageBatchGetRequest}.
+     * <p>
+     * This API is currently EXPERIMENTAL and is subject to change.
+     *
+     * @param streamName the name of the stream
+     * @param messageBatchGetRequest the request details
+     * @return a queue used to asynchronously receive {@link MessageInfo}
+     * @throws IOException covers various communication issues with the NATS
+     *         server such as timeout or interruption
+     * @throws JetStreamApiException the request had an error related to the data
+     */
+    LinkedBlockingQueue<MessageInfo> iterateMessageBatch(String streamName, MessageBatchGetRequest messageBatchGetRequest) throws IOException, JetStreamApiException;
+
+    /**
+     * Request a batch of messages using a {@link MessageBatchGetRequest}.
+     * <p>
+     * This API is currently EXPERIMENTAL and is subject to change.
+     *
+     * @param streamName the name of the stream
+     * @param messageBatchGetRequest the request details
+     * @param handler the handler used for receiving {@link MessageInfo}
+     * @throws IOException covers various communication issues with the NATS
+     *         server such as timeout or interruption
+     * @throws JetStreamApiException the request had an error related to the data
+     */
+    void gatherMessageBatch(String streamName, MessageBatchGetRequest messageBatchGetRequest, MessageInfoHandler handler) throws IOException, JetStreamApiException;
```

Comment:
I got rid of "gather" and just did "request" so as not to make it seem like I only support "scatter gather".  I also changed "iterate" to "queue" since it seems more accurate and also since we aren't giving them an iterator.

Reply:
Done :ok_hand: